packages/seacas/libraries/ioss/src/Ioss_Beam2.C

```diff
@@ -62,6 +63,16 @@ Ioss::Beam2::Beam2() : Ioss::ElementTopology(Ioss::Beam2::name, "Beam_2")
   Ioss::ElementTopology::alias(Ioss::Beam2::name, "line");
   Ioss::ElementTopology::alias(Ioss::Beam2::name, "line2");
   Ioss::ElementTopology::alias(Ioss::Beam2::name, "BEAM_2");
+
+  permutation()->alias(get_aliases(Ioss::ElementTopology::name()));
+}
+
+Ioss::ElementPermutation *Ioss::Beam2::permutation() const
+{
+  auto perm = Ioss::ElementPermutation::factory(Ioss::LinePermutation::name);
```

Comment:
Yes, I initially thought of an implementation based on the topology shape and you are right that it would simplify a lot of the code because it would all be in the base class. I like the `base_topology_name` idea because it handles regular and super elements. I'll revise based on this